app/jobs/regular/chat_channel_delete.rb

```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+
+module Jobs
+  class ChatChannelDelete < ::Jobs::Base
+    def execute(args = {})
+      chat_channel = ::ChatChannel.with_deleted.find_by(id: args[:chat_channel_id])
+
+      # this should not really happen, but better to do this than throw an error
+      if chat_channel.blank?
+        Rails.logger.warn("Chat channel #{args[:chat_channel_id]} could not be found, aborting delete job.")
+        return
+      end
+
+      DistributedMutex.synchronize(
+        "delete_chat_channel_#{chat_channel.id}",
+        validity: 20.minutes
+      ) do
+        Rails.logger.debug("Deleting webhooks and events for channel #{chat_channel.id}")
+        ChatMessage.transaction do
+          webhooks = IncomingChatWebhook.where(chat_channel: chat_channel)
+          ChatWebhookEvent.where(incoming_chat_webhook_id: webhooks.pluck(:id)).delete_all
```

Comment:
```suggestion
          ChatWebhookEvent.where(incoming_chat_webhook_id: webhooks.select(:id)).delete_all
```

This will allow us to run a single SQL query instead of 2.